src/dysh/spectra/spectrum.py

```diff
@@ -542,23 +542,26 @@ def shift(self, s, remove_wrap=True, fill_value=np.nan, method="fft"):
             "fft" uses a phase shift.
         """
 
-        new_data = core.data_shift(self.data, s, remove_wrap=remove_wrap, fill_value=fill_value, method=method)
+        new_spec = self._copy()
+        new_data = core.data_shift(new_spec.data, s, remove_wrap=remove_wrap, fill_value=fill_value, method=method)
 
         # Update data values.
-        self._data = new_data
+        new_spec._data = new_data
 
         # Update metadata.
-        self.meta["CRPIX1"] += s
+        new_spec.meta["CRPIX1"] += s
```

Comment:
Once we have flagging (masking) in place, we will also want to shift the mask?  But then have to decide what to do for non-integral channel shifts.

Reply:
Yes, we will have to.

Reply:
we can test, maybe we'll have to round the fractional shifts.